wikimedia/wikipedia-ios
# Peek and pop for lead image

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T253270

### Notes
* This only works on iOS 13 and higher. I think this broken behavior on iOS 12 is related to [this ticket](https://phabricator.wikimedia.org/T253330), so I focused on iOS 13 in this fix.
* This fix works on iPad - the lead image pops into the gallery controller. Notably, iPad on iOS 13+ has a bug for other (non-lead) images in articles - when coming from the webview, images peek but don't pop. I filed a bug for this: https://phabricator.wikimedia.org/T258109
* If no image has been opened on the article, the mediaList for the article is empty - and we need to fake the media list. (And to do that, we needed to write explicit initializers for all the codable structs.) This also means that there is a slight flash back to the article between peeking and popping. It doesn't seem worthwhile to dig into this further (given that this is supposed to be a quick fix), but am happy to dig in further (or hear ideas on how to improve it) if anyone feels like it is warranted.

### Test Steps
Test on both iPhone and iPad
1. Open an article that is not cached and has a lead image.
1. Long press the lead image. It should open in a peek.
1. Tap the image again. It should pop into gallery view.
1. Long press the lead image again. It should again open in a peek. Tap outside the peekable VC. The peek should close, and you should again be on an article.
1. Tap on the lead image, so you're on the gallery view. (This ensures the media list has been loaded.)
1. Long press the lead image. It should open in a peek.
1. Tap the image again. It should pop into gallery view.

### Screenshots/Videos *(if applicable)*
<img width="300" alt="Screenshot 10" src="https://user-images.githubusercontent.com/9295855/87596349-cd541f80-c6a4-11ea-9504-6c327b87b21b.png">

<img width="150" alt="Screenshot 10" src="https://user-images.githubusercontent.com/9295855/87596369-d1803d00-c6a4-11ea-81c8-3ec89e32c3f4.png"><img width="150" alt="Screenshot 10" src="https://user-images.githubusercontent.com/9295855/87596371-d218d380-c6a4-11ea-8352-00f76a2a566f.png">
